kosaraju_Algorithm.cpp

```diff
@@ -0,0 +1,97 @@
+//Strongly connected components(KOSARAJU ALGORITHM)
+#include <iostream>
+#include <vector>
+#include <stack>
+#include <unordered_set>
+using namespace std;
+//dfs to store vertices in a stack as per their finish time
+void dfs(vector<int>* edges, int start, unordered_set<int> &visited, stack<int> &finishStack) { 
+    visited.insert(start);
+    for (int i = 0; i < edges[start].size(); i++) {
+        int adjacent = edges[start][i];
+        if (visited.count(adjacent) == 0) {
+            dfs(edges, adjacent, visited, finishStack);
+        }
+    }
+    finishStack.push(start);
+}
+
+void dfs2(vector<int>* edges, int start, unordered_set<int>* component, unordered_set<int> & visited) { //dfs of a transpose graph
+	visited.insert(start);
+	component->insert(start);
+	for (int i = 0; i < edges[start].size(); i++) {
+		int adjacent = edges[start][i];
+		if (visited.count(adjacent) == 0) {
+			dfs2(edges, adjacent, component, visited);
+		}
+	}
+}
+
+unordered_set<unordered_set<int>*>* getSCC(vector<int>* edges, vector<int>* edgesT, int n) {
+    unordered_set<int> visited;
+    stack<int> finishedVertices;
+    for (int i = 0; i < n; i++) {
+        if (visited.count(i) == 0) {
+            dfs(edges, i, visited, finishedVertices);
+        }
+    }
+    unordered_set<unordered_set<int>*>* output = new unordered_set<unordered_set<int>*>();
+    visited.clear();
+    while (finishedVertices.size() != 0) {
+        int element = finishedVertices.top();
+        finishedVertices.pop();
+        if (visited.count(element) != 0) {
+            continue;
+        }
+        unordered_set<int>* component = new unordered_set<int>();
+        dfs2(edgesT, element, component, visited);
+        output->insert(component);
+    }
+    return output;
+}
+int main() {
+    int n;                                             //no.of vertices
+    cout<<"Enter the no. of vertices : "<<endl;
+    cin >> n;
+    vector<int>* edges = new vector<int>[n];           //using a adjacency list to implement
+    vector<int>* edgesT = new vector<int>[n];         //transpose of a graph
+    int m;                                             // no.of edges
+    cout<<"Enter the no. of edges : "<<endl;
+    cin >> m;
```

Comment:
keep a line space

Reply:
@asha15  i didn't get u can u please say what u want to say by "keep a line space"

Reply:
keep a blank line after this line